includes/subscriptions/class-wc-payments-subscription-migration-log-handler.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Class WC_Payments_Migration_Log_Handler
+ *
+ * @package WooCommerce\Payments
+ */
+
+use Automattic\Jetpack\Constants;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class for handling Stripe billing -> tokenized migration logs.
+ */
+class WC_Payments_Subscription_Migration_Log_Handler {
+
+	/**
+	 * Log handle (name).
+	 *
+	 * @var string
+	 */
+	const HANDLE = 'woo-payments-subscription-migration';
+
+	/**
+	 * A flag temporarily stored in the context column used to identify DB log entries that have been extended to avoid automated deletion.
+	 *
+	 * @var array
+	 */
+	const EXTENDED_DB_ENTRY_FLAG = 'extended_migration_log';
+
+	/**
+	 * The number of years to extend the life of a DB log entry for temporarily.
+	 *
+	 * @var int
+	 */
+	const DB_ENTRY_EXTENSION_IN_YEARS = 5;
+
+	/**
+	 * The holding property for our WC_Logger instance.
+	 *
+	 * @var WC_Logger
+	 */
+	private static $logger = null;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		// WC Core will delete logs on priority 10, so we need to run before that.
+		if ( $this->has_file_logger_enabled() ) {
+			add_action( 'woocommerce_cleanup_logs', [ $this, 'extend_life_of_migration_file_logs' ], 5 );
+		} elseif ( $this->has_db_logger_enabled() ) {
+			add_action( 'woocommerce_cleanup_logs', [ $this, 'extend_life_of_migration_db_logs' ], 5 );
+			add_action( 'woocommerce_cleanup_logs', [ $this, 'restore_db_log_timestamps' ], 100 );
+		}
+	}
+
+	/**
+	 * Logs a message to the migration log.
+	 *
+	 * @param string $message The message to log.
+	 */
+	public static function log( $message ) {
```

Comment:
I'm guessing this function is static so that it can be called within the migrator class being introduced by https://github.com/Automattic/woocommerce-payments/pull/6706?

After reviewing this PR, I've been thinking about how our two classes will interact and rather instantiate the class in the main `class-wc-payments.php` file, what are your thoughts on loading this class and instantiating it inside the subscriptions migrator class?

With this approach, the migrator construct will look like:

```
public function __construct( $api_client = null ) {
	$this->api_client = $api_client;
	$this->logger     = new WC_Payments_Subscription_Migration_Log_Handler();

	// Hook onto Scheduled Action to migrate wcpay subscription.
	add_action( 'wcpay_migrate_subscription', [ $this, 'migrate_wcpay_subscription' ] );
}
```

And then we can remove the mocked `log()` function in the migrator class and call:

```
$this->logger->log( 'message' );
```

Reply:
I think that makes a lot of sense. IIRC I defined this statically so I could fairly easily test writing to the log file without having to instantiate a new instance. I think using a static variable also made things a little simpler for unit tests.

Reply:
Unfortunately changing this to a non-static function meant I had to make a number of changes so I could unit test it. Hopefully this approach still works for you.